n8n-io/n8n
# 20200925 release

# Details

Created a ClockifyWriter node that allows creation of projects, tags and time entries through the Clockify API.

# Steps to test changes

1. Run "npm run dev" to view the website on your localhost.
1. Make the nodes 2 different ways.
>Trigger and Regular Node
>>1. Search the trigger nodes for "Clockify Trigger"
>>1. Make the node and enter your Clockify API key and pick the workspace you wish to pull time entry information from.
>>1. Drag the black dot on the Clockify Trigger node to make an attached regular node and search "Clockify".
>>1. Enter your API key again (or a different API key), pick the "Create" operation and either "Project", "Tag" or "Time Entry" for resource.
>>1. The Workspace should pull in a list of associated workspaces from the API key, and Client should pull in a list of clients associated with he chosen Workspace. Same for the User, Project, Tags and Task drop downs.
>>1. Clock "parameter Options", then "Add Expression to choose a value brought in by the attached Clockify Trigger node.
>>>For Project, choose Current Node>Input Data>JSON>project>name
>>>For Project Name, choose Current Node>Input Data>JSON>project>name
>>>For Billable, choose Current Node>Input Data>JSON>project>billable
>>>For Task, choose Current Node>Input Data>JSON>task
>>>For Tags, choose Current Node>Input Data>JSON>tags>Raw Value
>>>For Tag Name, choose Current Node>Input Data>JSON>tags>Raw Value
>>>For Start, choose Current Node>Input Data>JSON>timeInterval>start
>>>For End, choose Current Node>Input Data>JSON>timeInterval>end
>>>For Description, choose Current Node>Input Data>JSON>description
>>>For Project Color, choose Current Node>Input Data>JSON>project>color
>>>For Public, choose Current Node>Input Data>JSON>project>public
>>>For Note, choose Current Node>Input Data>JSON>project>note

>Just Regular Node
>>1. Make a new regular node by searching "Clockify" and picking that one.
>>1. Enter your API key, pick the "Create" operation and either "Project", "Tag" or "Time Entry" for resource.
>>1. The Workspace should pull in a list of associated workspaces from the API key, and Client should pull in a list of clients associated with he chosen Workspace. Same for the User, Project, Tags and Task drop downs.
>>1. Type in the required information for whatever resource you are making.
3. After all the information is in, click "Execute Node" and it should write all the information of the chosen resource type from the Workspace you chose in the "Clockify Trigger" to the workspace you chose in the regular "Clockify" node based on the pulled in Time Entries from the trigger.
1. The results should be returned to the node after it is done writing. You can read them there or go to that workspaces [Clockify Tracker page](https://clockify.me/tracker) page to check if everything was successful..
